src/handler.js

```diff
@@ -49,31 +52,44 @@ async function renderVideo(replay, callback, forceUUID = null) {
       headers: HEADERS
     });
   } catch (error) {
+    segment.addError(error);
     callback(null, {
       statusCode: 500,
       body: JSON.stringify(error)
     });
+  } finally {
+    segment.close();
   }
 }
 
 module.exports.render = async (event, context, callback) => {
+  const segment = new AWSXRay.Segment('render');
   const replayJSON = event.body;
   const replay = JSON.parse(replayJSON);
   if (replay.log && replay.id) {
-    await renderVideo(replay.log, callback, replay.id);
+    await renderVideo(replay.log, callback, segment, replay.id);
   } else {
-    await renderVideo(replay, callback);
+    await renderVideo(replay, callback, segment);
   }
 };
 
 module.exports.renderFromS3 = async (event, context, callback) => {
+  const segment = new AWSXRay.Segment('renderFromS3');
   const srcBucket = event.Records[0].s3.bucket.name;
   const srcKey = decodeURIComponent(event.Records[0].s3.object.key.replace(/\+/g, " "));
   const tmpPath = await download_and_return_tmp_path(srcBucket, srcKey);
   const replayJSON = fs.readFileSync(tmpPath);
   fs.unlinkSync(tmpPath);
-  const replay = JSON.parse(replayJSON);
-  await renderVideo(replay, callback, srcKey.replace(`${SOURCE_KEY}/`, ''));
+  let replay = [];
+  try {
```

Comment:
Do we want to keep this try / catch block? There doesn't seem to be a lot of other error handling in this handler, which is ok as long as we get the stacktrace in the logs. Do you know if we do? If so, I'd say remove it.

Also, either way, it looks like we're swallowing the exception here, which doesn't seem right.